pkg/proxy/proxy.go

```diff
@@ -503,11 +682,16 @@ func extractUserToken(req *http.Request) (string, error) {
 func (p *Proxy) newReverseProxy(ctx echo.Context, target *access.ClusterAccess, isPlugin bool) *httputil.ReverseProxy {
 	req := ctx.Request()
 	targetQuery := target.APIURL().RawQuery
+	username, _ := ctx.Get(context.UsernameKey).(string)
+	ctx.Set(context.ImpersonateUser, target.Username())
```

Comment:
minor - should we add a comment explaining why we need this ? I guess it's just for logging purposes right ?

Reply:
good idea, done in 599e2d2424b3fb2f5752ad57e7aef0496fe61783